### PR TITLE
Add tests of tracing SystemC model verilated with --hierarchical

### PR DESCRIPTION
**Pull request by yTakatsukasa**
https://github.com/verilator/verilator/blob/30ccccdb67629852dfe427be63a47bf86fb2e8c8/examples/make_tracing_sc/sc_main.cpp#L83-L85
As noted in the example, the following order must be kept.
1. Construct SystemC modules and connect them
2. finish elaboration by calling sc_sim_context::elaborate() or sc_start(SC_ZERO_TIME)
3. Pass Verilated(Vcd|Fst)Sc to trace()
4. Open the trace file

This PR adds
- test
- checks
- sc_start invocation to driver.pl

As usual, push tests and checks, then push the fix later.

### CODE REVIEW

File: include/verilated_fst_sc.cpp [hunk 1]
```diff
@@ -27,6 +27,17 @@
 //======================================================================
 //======================================================================
 
+void VerilatedFstSc::open(const char* filename) {
+    if (!sc_core::sc_get_curr_simcontext()->elaboration_done()) {
+        std::cerr << "VerilatedFstSc::open(\"" << filename
```
Comment (wsnyder) [inline include/verilated_fst_sc.cpp:32]:
```suggestion
        std::cerr << "%Error: VerilatedFstSc::open(\"" << filename
```

But perhaps vl_fatal is better.

File: include/verilated_vcd_sc.cpp [hunk 2]
```diff
@@ -27,6 +27,17 @@
 //======================================================================
 //======================================================================
 
+void VerilatedVcdSc::open(const char* filename) {
+    if (!sc_core::sc_get_curr_simcontext()->elaboration_done()) {
+        std::cerr << "VerilatedVcdSc::open(\"" << filename
```
Comment (wsnyder) [inline include/verilated_vcd_sc.cpp:32]: ditto from above.

File: src/V3EmitCModel.cpp [hunk 3]
```diff
@@ -581,6 +581,14 @@ class EmitCModel final : public EmitCFunc {
         // ::trace
         puts("\nVL_ATTR_COLD void " + topClassName() + "::trace(");
         puts(v3Global.opt.traceClassBase() + "C* tfp, int levels, int options) {\n");
+        if (optSystemC()) {
+            puts(/**/ "if (!sc_core::sc_get_curr_simcontext()->elaboration_done()) {\n");
+            puts(/****/ "std::cerr << \"" + topClassName()
+                 + "::trace() is called before sc_core::sc_start(sc_core::SC_ZERO_TIME)\" << "
+                   "std::endl;\n");
+            puts(/****/ "std::abort();\n");
```
Comment (wsnyder) [inline src/V3EmitCModel.cpp:589]: Think vl_fatal is better since called as part of model.

**Review by wsnyder — COMMENTED** [review]
Makes sense - just some notes about how to error.

**Comment by yTakatsukasa** [comment]
Merge this PR.
Please let me know if you find something.

**Pull request merged**